src/Statement/StatementList.php

```diff
@@ -45,10 +45,20 @@ public function __construct( $statements = array() /*...*/ ) {
 			$statements = func_get_args();
 		}
 
+		$this->setStatements( $statements );
+	}
+
+	/**
+	 * @param Statement[]|Traversable
+	 *
+	 * @throws InvalidArgumentException
+	 */
+	public function setStatements( $statements ) {
 		if ( !is_array( $statements ) && !( $statements instanceof Traversable ) ) {
 			throw new InvalidArgumentException( '$statements must be an array or an instance of Traversable' );
 		}
 
+		$this->statements = array();
```

Comment:
I'm wondering about the position of this relative to the throwing of extensions. If the method now gets say, a list of ints, the statement list will be cleared and then an exception will be thrown. I can't think of a concrete case we are likely to run into where this will be a problem, though do wonder if this is something to best avoid.